### PR TITLE
FT-47 APIs for feed: Ideas, queries, support ticket, galleries

### PR DESCRIPTION
**Pull request by sam-ludlow**
https://io1dev.atlassian.net/browse/FT-47

### CODE REVIEW

File: api.swagger.yaml [hunk 1]
```diff
@@ -2974,6 +3070,110 @@ paths:
...
+            type: string
+            example: '2012-12-25'
+        - name: updated_date
+          description: 'Filter on date updated. Format: YYYY-MM-DD.'
+          in: query
+          schema:
+            type: string
+            example: '2012-12-25'
+        - name: state
+          description: 'Filter the list of lists on a comma seperated list of states. Valid statuses are "draft", "published", and "unpublished". You may include multiple. The default when not supplied is "published" only.'
```
Comment (ParryLuke) [inline api.swagger.yaml:3099]: ideas/queries/support tickets don't use workflow states

File: api.swagger.yaml [hunk 2]
```diff
@@ -11175,27 +11583,207 @@ components:
                   format: uuid
                   example: '01234567-89ab-cdef-1234-56789abcdef0'
             team:
-              description: 'The team targeting this custom content.'
+              description: 'The team targeting this custom content.'
```
Comment (ParryLuke) [inline api.swagger.yaml:11586]: *custom content typo

There are quite a few of these, could the descriptions be reviewed?

File: api.swagger.yaml [hunk 3]
```diff
@@ -14511,6 +15200,171 @@ components:
...
+      allOf:
+        - $ref: '#/components/schemas/query-schema-common'
+        - type: object
+          properties:
+            uuid:
+              description: 'Invotra UUID of the query.'
+              type: string
+              format: uuid
+              example: '01234567-89ab-cdef-1234-56789abcdef0'
+            attachment:
```
Comment (ParryLuke) [inline api.swagger.yaml:15213]: queries do not have attachments. Could we double check all of the fields in the schemas in this PR exist on the CT? And in solr, as we could exclude them from the task if not, we only need the basic endpoint for now

File: api.swagger.yaml (hunk 2 above)
Comment (ParryLuke) [reply api.swagger.yaml:11586]: I just realised this is to do with webforms anyway, which is not required for this task.

**Review by ParryLuke — CHANGES_REQUESTED** [review]

File: api.swagger.yaml (hunk 1 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: took out for them 3.

File: api.swagger.yaml (hunk 2 above)
Comment (sam-ludlow) [reply api.swagger.yaml:11568]: Pretty sure I've got all the "custom content" typos out, they where all over at one point.

File: api.swagger.yaml (hunk 3 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: removed

File: api.swagger.yaml [hunk 4]
```diff
@@ -14511,6 +15150,124 @@ components:
...
+              example: 1510941596
+            timestamp_updated:
+              description: 'This is the timestamp (in seconds) of when the support ticket was last updated.'
+              type: integer
+              example: 1510941696
+            comment_count:
+              description: 'Count of comments on the support ticket.'
+              type: integer
+              example: 5
+            likes_count:
```
Comment (ParryLuke) [inline api.swagger.yaml:15248]: Support tickets don't have likes, can you see this available in solr? I was checking in the drupal UI

File: api.swagger.yaml (hunk 2 above)
Comment (ParryLuke) [reply api.swagger.yaml:11568]: Are custom content changes expected in this PR?

**Review by ParryLuke — CHANGES_REQUESTED** [review]

File: api.swagger.yaml (hunk 4 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: done

**Review by ParryLuke — APPROVED** [review]

**Comment by MireilleHelle** [comment]
Feeds project Swagger changes were merged under https://github.com/invotra/api/pull/343. PR no longer required. Closing.

**Pull request closed without merging**